pkg/asana/scim.go

```diff
@@ -0,0 +1,318 @@
+
+	var result ScimListResponse
+	var scimError ScimError
+	resp, err := c.httpClient.Do(req,
+		uhttp.WithJSONResponse(&result),
+		uhttp.WithErrorResponse(&scimError),
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	return &result, nil
+}
+
+// GetScimUser gets a specific user by ID via the SCIM API.
+func (c *Client) GetScimUser(ctx context.Context, userID string) (*ScimUser, error) {
+	userUrl, err := getPath(ScimBaseUrl, fmt.Sprintf("/Users/%s", userID))
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := c.httpClient.NewRequest(
+		ctx,
+		http.MethodGet,
+		userUrl,
+		uhttp.WithBearerToken(c.accessToken),
+		uhttp.WithAcceptJSONHeader(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var result ScimUser
+	var scimError ScimError
+	resp, err := c.httpClient.Do(req,
+		uhttp.WithJSONResponse(&result),
+		uhttp.WithErrorResponse(&scimError),
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	return &result, nil
+}
+
+// CreateScimUser creates a new user via the SCIM API.
+func (c *Client) CreateScimUser(ctx context.Context, user *ScimUser) (*ScimUser, error) {
+	userUrl, err := getPath(ScimBaseUrl, "/Users")
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure schemas are set
+	if user.Schemas == nil {
+		user.Schemas = []string{scimUserSchema}
+	}
+
+	// If there's enterprise extension data, add that schema
+	if user.EnterpriseExtension != (ScimEnterpriseExtension{}) {
+		hasEnterpriseSchema := false
+		for _, schema := range user.Schemas {
+			if schema == scimEnterpriseUserSchema {
+				hasEnterpriseSchema = true
+				break
+			}
+		}
+		if !hasEnterpriseSchema {
+			user.Schemas = append(user.Schemas, scimEnterpriseUserSchema)
+		}
+	}
+
+	req, err := c.httpClient.NewRequest(
+		ctx,
+		http.MethodPost,
+		userUrl,
+		uhttp.WithBearerToken(c.accessToken),
+		uhttp.WithJSONBody(user),
+		uhttp.WithContentTypeJSONHeader(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var result ScimUser
+	var scimError ScimError
+	resp, err := c.httpClient.Do(req,
+		uhttp.WithJSONResponse(&result),
+		uhttp.WithErrorResponse(&scimError),
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	return &result, nil
+}
+
+// UpdateScimUser updates a user via the SCIM API.
+func (c *Client) UpdateScimUser(ctx context.Context, userID string, user *ScimUser) (*ScimUser, error) {
+	userUrl, err := getPath(ScimBaseUrl, fmt.Sprintf("/Users/%s", userID))
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure schemas are set
+	if user.Schemas == nil {
+		user.Schemas = []string{scimUserSchema}
+	}
+
+	// If there's enterprise extension data, add that schema
+	if user.EnterpriseExtension != (ScimEnterpriseExtension{}) {
+		hasEnterpriseSchema := false
+		for _, schema := range user.Schemas {
+			if schema == scimEnterpriseUserSchema {
+				hasEnterpriseSchema = true
+				break
+			}
+		}
+		if !hasEnterpriseSchema {
+			user.Schemas = append(user.Schemas, scimEnterpriseUserSchema)
+		}
```

Comment:
Only add schema if there is an enterpriseSchema, right?

```suggestion
		if hasEnterpriseSchema {
			user.Schemas = append(user.Schemas, scimEnterpriseUserSchema)
		}
```